keps/sig-node/3619-supplemental-groups-policy/README.md

```diff
@@ -92,37 +94,68 @@ Items marked with (R) are required *prior to targeting to a milestone / release*
+
+Moreover, this causes security considerations as follows. When a cluster enforces some security policy for pods that protects the value of `RunAsGroup` and `SupplementalGroups`, the effect of its enforcement is limited, i.e., cluster users can easily bypass the policy enforcement just by using a custom image. If such a bypass happened, it would be unexpected behavior for most cluster administrators because the enforcement is almost useless. Moreover, the bypass will cause unexpected file access permission. In some use cases, the unexpected file access permission will be a security concern. For example, using `hostPath` volumes could be a severe problem because UID/GIDs matter in accessing files/directories in the volumes.
 
-However, Kubernetes provides no API surface to prevent this bypass, although it could sometimes lead to a security concern. Because the behavior is implemented in CRI implementations actually, To mitigate this, the cluster administrators will need to deploy a custom low-level container runtime(e.g., [pfnet-research/strict-supplementalgroups-container-runtime](https://github.com/pfnet-research/strict-supplementalgroups-container-runtime)) that modifies OCI container runtime spec(`config.json`) produced by CRI implementations (e.g., containerd, cri-o). A custom `RuntimeClass` would be introduced for it. Nevertheless, It would be an extra operational burden for cluster administrators.
+Kubernetes provides no API surface to prevent this bypass, although it could sometimes lead to a security concern. Because the behavior is implemented in CRI implementations actually, To mitigate this, the cluster administrators will need to deploy a custom low-level container runtime(e.g., [pfnet-research/strict-supplementalgroups-container-runtime](https://github.com/pfnet-research/strict-supplementalgroups-container-runtime)) that modifies OCI container runtime spec(`config.json`) produced by CRI implementations (e.g., containerd, cri-o). A custom `RuntimeClass` would be introduced for it. Nevertheless, It would be an extra operational burden for cluster administrators.
 
-Thus, this KEP proposes to offer a new API field named `SupplementalGroupsPolicy` that customize to respect/ignore the group membership of the primary UID of the container defined in the container image when applying `SupplementalGroups` to the container processes. The new API allows cluster administrators to deploy security policies that protect the `SupplementalGroupsPolicy` field in the cluster to avoid the unexpected bypass of `SupplementalGroups` described above.
+Thus, this KEP proposes to offer a new API field named `SupplementalGroupsPolicy` that enables OCI-compliant behavior for supplemental groups to the first container process. The new API allows cluster administrators to deploy security policies that protect the `SupplementalGroupsPolicy` field in the cluster to avoid the unexpected bypass of `SupplementalGroups` described above.
```

Comment:
see above comment - let's make a specific decision about whether OCI-compliance is the goal, or whether the "k8s API is authoritative when used" principle is the goal.

Reply:
see above I don't think oci compliance covers client side config.json editing.. rather it covers how the final config.json will be processed.

Reply:
> let's make a specific decision about whether OCI-compliance is the goal, or whether the "k8s API is authoritative when used" principle is the goal.

I lean to the latter. see https://github.com/kubernetes/enhancements/pull/3620#discussion_r1098609784